src/sv-pipeline/scripts/make_evidence_qc_table.py

```diff
@@ -0,0 +1,269 @@
+    Args:
+        filename: a tab-delimited file containing wgd scores.
+    Returns:
+        A pandas DataFrame containing wgd scores for each sample.
+    """
+    df_wgd_scores = pd.read_csv(filename, sep="\t")
+    df_wgd_scores.rename(columns={'score': 'wgd_score'}, inplace=True)
+    return df_wgd_scores
+
+
+def read_non_diploid(filename: str) -> pd.DataFrame:
+    """
+    EvidenceQC → untar → /ploidy_est/binwise_CNV_qValues.bed.gz →
+    count number of bins with q-value < 0.05 for each sample
+    Args:
+        filename: a tab-delimited file containing binwise CNV qValues.
+    Returns:
+        A pandas DataFrame containing the number of bins with q-value < 0.05 for each sample.
+    """
+    df_non_diploid = pd.read_csv(filename, sep="\t").T
+    df_non_diploid = df_non_diploid.T.iloc[0:, 3:]
+    nondiploid_counts = df_non_diploid[df_non_diploid < 0.05].count()
+    nondiploid_counts_df = pd.DataFrame(nondiploid_counts, columns=["nondiploid_bins"]).rename_axis(
+        ID_COL).reset_index()
+    return nondiploid_counts_df
+
+
+def read_melt_insert_size(filename: str, col_name="mean_insert_size") -> pd.DataFrame:
+    """
+    Reads MELT insert size into a Pandas DataFrame.
+    Args:
+        filename: melt_insert_size in sample data table
+        col_name: Set title to use for the column containing MELT mean insert size.
+    Returns:
+        MELT insert size for each sample given in a Pandas DataFrame with two columns: sample ID and mean insert size.
+    """
+    columns_names = [ID_COL, col_name]
+    if filename is None:
+        return pd.DataFrame(colums=columns_names)
+    df = pd.read_csv(filename, names=columns_names, header=0, sep="\t")
+    df[col_name] = df[col_name].astype(float)
+    return df
+
+
+def get_col_name(caller: str, outlier_type: str) -> str:
+    return f"{caller}_{outlier_type}_outlier"
+
+
+def read_outlier(filename: str, outlier_col_label: str) -> pd.DataFrame:
+    """
+    Args:
+        filename: A tab-delimited file containing the output of
+        Evidence QC on a caller (e.g., Manta) with low/high.
+        outlier_col_label: Column header for the outliers.
+    Returns:
+        A pandas DataFrame containing the number of times a sample
+        appears in the QC of a caller output.
+    """
+    if filename is None:
+        return pd.DataFrame(columns=[ID_COL, outlier_col_label])
+    df = pd.read_csv(filename, sep="\t")
+    if df["Outlier_Sample"].empty:
+        df[ID_COL] = df.apply(lambda _: EMPTY_OUTLIERS, axis=0)
+        outlier_sample = df.pivot_table(columns=[ID_COL], aggfunc="size").astype(int)
+    else:
+        df['Outlier_Sample'] = df['Outlier_Sample'].apply(Path)
+        df[ID_COL] = df['Outlier_Sample'].apply(lambda x: x.stem if '.' in x.suffix else np.nan)
+        outlier_sample = df.pivot_table(columns=[ID_COL], aggfunc="size").astype(int)
+    outlier_df = outlier_sample.reset_index()
+    outlier_df.columns = [ID_COL, outlier_col_label]
+    return outlier_df
```

Comment:
Not necessary to merge / question of preference. I think my preference would be to fill cells with 0, rather than `np.nan`, if the sample does not appear as an outlier at all in a specified category. This would allow the outlier columns to be stored as ints and may make downstream computation a bit simpler because users won't have to account for NaN values. But, I recognize that it's easier to have NaN values due to the dataframe merge later on, and I'm open to other views.